eox_nelp/user_profile/required_fields_validation.py

```diff
@@ -0,0 +1,354 @@
+        else:
+            errors = validate_field(value, rules)
+
+        result[field] = errors
+
+    return result
+
+
+def validate_field(value, rules):
+    """
+    Validates a field based on its rules.
+
+    Args:
+        value (str): The field value to validate.
+        rules (dict): The validation rules.
+
+    Returns:
+        list: A list of validation error messages. Empty if valid.
+    """
+    errors = []
+
+    validators = {
+        "max_length": validate_max_length,
+        "char_type": validate_char_type,
+        "format": validate_format,
+        "optional_values": validate_optional_values,
+    }
+
+    for rule, argument in rules.items():
+        validator = validators.get(rule)
+
+        if validator and not validator(value, argument):
+            errors.append(f"{rule} with argument {argument} failed")
+
+    return errors
+
+
+def validate_max_length(value, max_length):
+    """
+    Checks if the given value does not exceed the specified max_length.
+
+    Args:
+        value (str): The value to be validated.
+        max_length (int): The maximum allowed length.
+
+    Returns:
+        bool: True if valid, False if it exceeds max_length.
+    """
+    return len(str(value)) <= max_length
+
+
+def validate_optional_values(value, allowed_values):
+    """
+    Checks if the given value is within the predefined allowed values.
+
+    Args:
+        value (str): The value to be validated.
+        allowed_values (list): A list of allowed values.
+
+    Returns:
+        bool: True if the value is in the allowed list, False otherwise.
+    """
+    return value in allowed_values
+
+
+def validate_char_type(value, char_type):
+    """
+    Validates if the given value matches the specified character type.
+
+    Args:
+        value (str): The value to be validated.
+        char_type (str): The character type to validate against. Options: "latin", "arabic", "cp1252".
+
+    Returns:
+        bool: True if the value matches the specified character type, False otherwise.
+    """
+    validators = {
+        "latin": validate_latin,
+        "arabic": validate_arabic,
+        "cp1252": validate_cp1252,
+    }
+
+    validator = validators.get(char_type)
+
+    if validator:
+        return validator(value)
+
+    return False
+
+
+def validate_latin(value):
+    """
+    Checks if the given value contains only Latin characters and standard punctuation.
+
+    Args:
+        value (str): The value to be validated.
+
+    Returns:
+        bool: True if valid, False otherwise.
+    """
+    latin_regex = r'^[a-zA-ZÀ-ÿ\s.,\'"-]*$'
+    return re.match(latin_regex, value) is not None
+
+
+def validate_arabic(value):
+    """
+    Checks if the given value contains only Arabic script characters.
+
+    Args:
+        value (str): The value to be validated.
+
+    Returns:
+        bool: True if valid, False otherwise.
+    """
+    arabic_regex = r'^[\u0600-\u06FF\s]+$'  # Unicode range for Arabic characters
```

Comment:
This could be imported from here?
https://github.com/nelc/nelp-custom-registration-fields/blob/main/custom_reg_form/forms.py#L26

Reply:
nope, basically that is a protected method